src/plugins/mer/mersdkdetailswidget.cpp

```diff
@@ -208,5 +233,131 @@ void MerSdkDetailsWidget::onPathChooserEditingFinished()
         emit sshKeyChanged(m_ui->privateKeyPathChooser->path());
 }
 
+void MerSdkDetailsWidget::onWwwProxyDisabledToggled(bool checked)
+{
+    if (!checked)
+        return;
+
+    m_wwwProxy = MER_SDK_PROXY_DISABLED;
+
+    m_ui->wwwProxyServersLabel->setHidden(true);
+
+    m_ui->wwwProxyServersLine->clear();
+    m_ui->wwwProxyServersLine->setHidden(true);
+
+    m_ui->wwwProxyExcludesLabel->setHidden(true);
+
+    m_ui->wwwProxyExcludesLine->clear();
+    m_ui->wwwProxyExcludesLine->setHidden(true);
+
+    emit wwwProxyChanged(m_wwwProxy, QString(), QString());
+}
+
+void MerSdkDetailsWidget::onWwwProxyAutomaticToggled(bool checked)
+{
+    if (!checked || m_wwwProxy == MER_SDK_PROXY_AUTOMATIC)
+        return;
+
+    m_wwwProxy = MER_SDK_PROXY_AUTOMATIC;
+
+    m_ui->wwwProxyServersLabel->setText(tr("URL:"));
+    m_ui->wwwProxyServersLabel->setHidden(false);
+
+    m_ui->wwwProxyServersLine->setText(m_wwwProxyServerUrl);
+    m_ui->wwwProxyServersLine->setHidden(false);
+    m_ui->wwwProxyServersLine->setToolTip(tr("Leave empty for asking the network, otherwise insert a proxy autoconfiguration URL."));
+
+    m_ui->wwwProxyExcludesLabel->setHidden(true);
+    m_ui->wwwProxyExcludesLine->clear();
+    m_ui->wwwProxyExcludesLine->setHidden(true);
+
+    emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServerUrl, QString());
+}
+
+void MerSdkDetailsWidget::onWwwProxyManualToggled(bool checked)
+{
+    if (!checked || m_wwwProxy == MER_SDK_PROXY_MANUAL)
+        return;
+
+    m_wwwProxy = MER_SDK_PROXY_MANUAL;
+
+    m_ui->wwwProxyServersLabel->setText(tr("Servers:"));
+    m_ui->wwwProxyServersLabel->setHidden(false);
+
+    m_ui->wwwProxyServersLine->setText(m_wwwProxyServerList);
+    m_ui->wwwProxyServersLine->setHidden(false);
+    m_ui->wwwProxyServersLine->setToolTip(tr("Insert a list of proxies. E.g. \"proxy1.example.com 1.2.3.4:8080\""));
+
+    m_ui->wwwProxyExcludesLabel->setHidden(false);
+
+    m_ui->wwwProxyExcludesLine->setText(m_wwwProxyExcludes);
+    m_ui->wwwProxyExcludesLine->setHidden(false);
+    m_ui->wwwProxyExcludesLine->setToolTip(tr("Insert a list of addresses excluded from the proxy. E.g. \"internal.example.com 10.0.0.0/8\""));
+
+    emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServerList, m_wwwProxyExcludes);
+}
+
+void MerSdkDetailsWidget::onWwwProxyServersEdited(const QString &s)
+{
+    QString proxies = s;
+    QRegularExpression reg;
+
+    int cpos = m_ui->wwwProxyServersLine->cursorPosition();
+    int l = proxies.length();
+
+    reg.setPattern("[,;\\s]");
+    if (m_wwwProxy == MER_SDK_PROXY_AUTOMATIC) {
+        // Allow to input a single item with no spaces, commas or semicolons
+        proxies.replace(reg, "");
```

Comment:
Wouldn't this break if the URL contains comma or semicolon? Sometimes these are used in tuple-like parameter values. Maybe cleaning whitespaces would be sufficient.

Reply:
Fixed by strong quoting in wwwproxycommand.cpp. All characters are now allowed for proxy autoconfiguration URL. In manual mode entries are separated by single spaces, so all characters except whitespaces are allowed for each entry.